### ISSUE TITLE
Displaying Relationship field takes too long to display when there is a lot of items

### ISSUE THREAD

**Issue by AntonioFernandezFernandez** [keystonejs/keystone#4797]
I have this model:

```
UserApp.add({
	employer_code: { type: Types.Text, label:'Código de empleado',initial: true, required: true, index: true },
	name: { type: Types.Text, label:'Nombre', required: true, index: true },
	email: { type: Types.Email,label:'Email', initial: true,unique: true, index: true },
	password: { type: Types.Password,label:'Contraseña', initial: true, required: true },
	locked: { type: Types.Boolean,label:'Bloqueado', default:true },
	failed_login_attempts: { type: Types.Number,label:'Inicios de sesión fallidos', default:0, noedit:false },
	last_connection: { type: Types.Datetime,label:'Última hora de conexión', default: Date.now, noedit:true },
	device_id: { type: Types.Text, label: 'Identificador de dispositivo', note:'Identificador único de dispositivo móvil'}
	
});

```

And another model with a relationship field of userapp

```
BaseItem.add({
    users: { type: Types.Relationship, ref: 'UserApp', refPath: '_id', label:'Enviar usuarios', many:true,  createInline: false }
});
```

When i load the BaseItem detail form with more than 100 userapp in users relationship, the relationship field takes too long for display
the user names in relationship field.

I inspected the console and saw these messages:

GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9ce?basic 200 2.726 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9cf?basic 200 11.684 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d0?basic 200 3.533 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d1?basic 200 4.630 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d2?basic 200 3.365 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d3?basic 200 3.355 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d5?basic 200 2.920 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d6?basic 200 5.366 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d4?basic 200 4.035 ms
GET /keystone/api/user-apps/5b98d2ed9254ae84ae8dd9d7?basic 200 2.972 ms
GET /keystone/api/user-apps/5b98d2ec9254ae84ae8dd963?basic 200 3.012 ms

Every record in users relationship makes a request to /keystone/api-user-apps/{i} for displaying the name in relationship field.
I think this is a very poor implementation because maybe, it can be solved with only one query.

¿any suggestions?

### Expected behavior
Relationship field may display results making only one query with objectid array as filter 

### Actual/Current behavior
Relationship field takes too long to display

### Steps to reproduce the actual/current behavior
Fill a relationship field with more than 50 items

### Environment

| Software         | Version
| ---------------- | -------
| Keystone         | 4.0.0
| Node.js          | NodeJS 6.12.3
| Browser          |

**Comment by AntonioFernandezFernandez** [comment]
In mi opinion this is not a question, its a bug, loading time it's unaceptable when have more than 50 items and it's absolutely unnecesary to make 50 request to load a simple relationship field.

**Comment by mcclowes** [comment]
Given that each of those requests is taking <12ms, surely that's not the issue?

**Comment by autoboxer** [comment]
Just adding a note that a duplicate of this issue #4891 has been closed to make this the sole tracking mechanism.  I'm also changing this to *bug* from *question* as I agree with @AntonioFernandezFernandez.

**Comment by chrisschaub** [comment]
Yeah, fetching each item with a "get" request when we already have the list of items seems way overkill. Wouldn't it be better to just do the limit here and render each one within the results object?

**Comment by chrisschaub** [comment]
Just to further expand on why this is an issue, the entire page is frozen until all of these relationship GET requests are made. I can take up to 10 seconds before the page becomes responsive on a larger site. Would a possible quick work around be to only fetch the suggestions when the field is clicked? That way, just loading the page doesn't incur the wait. It should be rewritten as a render in place for the items, but maybe it shouldn't be pre-fetching anything unless asked to do so?
